error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/BadStaticImport.java

```diff
@@ -0,0 +1,194 @@
+   * from this collection.
+   *
+   * <p>This should be mutually exclusive with {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS}
+   */
+  // XXX: Perhaps the set of exempted `java.util.Collections` methods is too strict. For now any
+  // method name that could be considered "too vague" or could conceivably mean something else in a
+  // specific context is left out.
+  static final ImmutableSetMultimap<String, String> BAD_STATIC_IMPORT_CANDIDATE_MEMBERS =
+      ImmutableSetMultimap.<String, String>builder()
+          .put("com.mongodb.client.model.Filters", "empty")
+          .putAll(
+              "java.util.Collections",
+              "addAll",
+              "copy",
+              "fill",
+              "list",
+              "max",
+              "min",
+              "nCopies",
+              "rotate",
+              "sort",
+              "swap")
+          .put("java.util.Locale", "ROOT")
+          .put("java.util.Optional", "empty")
+          .putAll("java.util.regex.Pattern", "compile", "matches", "quote")
+          .put("org.springframework.http.MediaType", "ALL")
+          .build();
+
+  /**
+   * Identifiers that should never be statically imported.
+   *
+   * <p>Identifiers listed by {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS} should be
+   * mutually exclusive with identifiers listed here.
+   *
+   * <p>This should be a superset of the identifiers flagged by {@link
+   * com.google.errorprone.bugpatterns.BadImport}.
+   */
+  static final ImmutableSet<String> BAD_STATIC_IMPORT_CANDIDATE_IDENTIFIERS =
+      ImmutableSet.of(
+          "builder",
+          "create",
+          "copyOf",
+          "from",
+          "getDefaultInstance",
+          "INSTANCE",
+          "MIN",
+          "MIN_VALUE",
+          "MAX",
+          "newBuilder",
+          "of",
+          "valueOf");
+
+  /** Instantiates a new {@link BadStaticImport} instance. */
+  public BadStaticImport() {}
+
+  @Override
+  public Description matchIdentifier(IdentifierTree tree, VisitorState state) {
+    if (isMatch(tree, state)) {
+      return getDescription(tree, state);
+    }
+
+    return Description.NO_MATCH;
+  }
+
+  private Description getDescription(IdentifierTree tree, VisitorState state) {
+    Symbol symbol = ASTHelpers.getSymbol(tree);
+    SuggestedFix.Builder builder =
+        SuggestedFix.builder().removeStaticImport(getImportToRemove(symbol));
+    String replacement =
+        SuggestedFixes.qualifyType(state, builder, symbol.getEnclosingElement()) + ".";
+    builder.prefixWith(tree, replacement);
+    return describeMatch(tree, builder.build());
+  }
+
+  @SuppressWarnings("NullAway")
+  private static String getImportToRemove(Symbol symbol) {
+    return String.join(
+        ".", symbol.getEnclosingElement().getQualifiedName(), symbol.getSimpleName());
+  }
+
+  private static boolean isMatch(IdentifierTree tree, VisitorState state) {
+    Symbol symbol = ASTHelpers.getSymbol(tree);
```

Comment:
We use this method here and in the `getDescription`. We can probably start with that as we are more interested in that than in the Tree itself. Saves us an extra if statement.